winml/adapter/WinMLAdapterErrors.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "core/common/status.h"
+#include "core/common/common.h"
+
+inline __declspec(noinline) winrt::hresult_error _winmla_to_hresult() noexcept {
+  try {
+    throw;
+  } catch (winrt::hresult_error const& e) {
+    return e;
+  } catch (wil::ResultException const& e) {
+    return winrt::hresult_error(e.GetErrorCode(), winrt::to_hstring(e.what()));
+  } catch (std::bad_alloc const&) {
+    return winrt::hresult_error(E_OUTOFMEMORY);
+  } catch (std::out_of_range const& e) {
+    return winrt::hresult_out_of_bounds(winrt::to_hstring(e.what()));
+  } catch (std::invalid_argument const& e) {
+    return winrt::hresult_invalid_argument(winrt::to_hstring(e.what()));
+  } catch (onnxruntime::OnnxRuntimeException const& e) {
+    return winrt::hresult_error(StatusCodeToHRESULT(e.GetStatusCode()));
```

Comment:
>StatusCodeToHRESULT [](start = 32, length = 19)

are we losing the string?   can we added that as well?  hresult_error lets you pass both a code and a string.